cirq-rigetti/cirq_rigetti/quil_input.py

```diff
@@ -234,50 +303,127 @@ def circuit_from_quil(quil: str) -> Circuit:
+            )
+            confusion_matrix = entries.reshape((2, 2)).T
+
+            """
+            Incompatible types in assignment
+                expression has type
+                    "ndarray[Any, dtype[complexfloating[Any, Any]]]"
+                target has type
+                    "ndarray[Any, dtype[floating[Any]]]"
+            """
+            confusion_maps[qubit] = confusion_matrix
+
+        else:
+            raise UnsupportedQuilInstruction(PRAGMA_ERROR)
+
+    # Interpret the instructions
+    for inst in program:
         # Pass when encountering a DECLARE.
-        elif isinstance(inst, Declare):
+        if isinstance(inst, Declare):
             pass
 
         # Convert pyQuil gates to Cirq operations.
         elif isinstance(inst, PyQuilGate):
             quil_gate_name = inst.name
             quil_gate_params = inst.params
-            line_qubits = list(LineQubit(q.index) for q in inst.qubits)
+            line_qubits = list(LineQubit(qubit_index(q)) for q in inst.qubits)
             if quil_gate_name not in defined_gates:
                 raise UndefinedQuilGate(f"Quil gate {quil_gate_name} not supported in Cirq.")
             cirq_gate_fn = defined_gates[quil_gate_name]
             if quil_gate_params:
-                circuit += cast(Callable[..., Gate], cirq_gate_fn)(*quil_gate_params)(*line_qubits)
+                params = [quil_expression_to_sympy(p) for p in quil_gate_params]
+                transformer = parameter_transformers[quil_gate_name]
+                circuit += cast(Callable[..., Gate], cirq_gate_fn)(**transformer(*params))(
+                    *line_qubits
+                )
             else:
                 circuit += cirq_gate_fn(*line_qubits)
 
         # Convert pyQuil MEASURE operations to Cirq MeasurementGate objects.
         elif isinstance(inst, PyQuilMeasurement):
-            line_qubit = LineQubit(inst.qubit.index)
+            qubit = qubit_index(inst.qubit)
+            line_qubit = LineQubit(qubit)
             if inst.classical_reg is None:
                 raise UnsupportedQuilInstruction(
                     f"Quil measurement {inst} without classical register "
                     f"not currently supported in Cirq."
                 )
             quil_memory_reference = inst.classical_reg.out()
-            circuit += MeasurementGate(1, key=quil_memory_reference)(line_qubit)
+            if qubit in confusion_maps:
+                cmap_key: Tuple[int, ...] = (qubit,)
+                cmap_value: NDArray[Any] = confusion_maps[qubit]
+                cmap = {cmap_key: cmap_value}
+                # cmap = {(qubit,): confusion_maps[qubit]}
+                """
+                Argument "confusion_map" to "MeasurementGate" has incompatible type
+                    "         Dict[Tuple[int],      ndarray[Any, dtype[floating[Any]]]]"
+                expected
+                    "Optional[Dict[Tuple[int, ...], ndarray[Any, Any]]]"
+                """
```

Comment:
I've fixed this error with explicit type definitions on `cmap_key` and `cmap_value`, but I'm not sure they're correct.

Reply:
Perfect, that fixed it, thanks!